### ISSUE TITLE
Ability to bypass creation of administration role when using env prefix in singledb mode

### ISSUE THREAD

**Issue by TedCha** [littleK0i/SnowDDL#135]
**Is your feature request related to a problem? Please describe.**
When running SnowDDL with the `--env-prefix` flag in singledb mode, an administration role tries to be created within [`context.py`](https://github.com/littleK0i/SnowDDL/blob/master/snowddl/context.py#L63). As a developer working in an environment where I only have granted permissions to specific databases and have no SYSADMIN capability, this stops me from using the environment prefix functionality.

**Describe the solution you'd like**
I'd like the ability to bypass the creation of the new administration role when using the `--env-prefix` flag within singledb mode. I understand there is a risk with correct OWNERSHIP and a SHOW limit but I believe this risk is mitigated in single DB mode where the number of objects the role owns is limited to a single database versus an entire account.

**Describe alternatives you've considered**
An alternative solution could be to create the administration role as a [database role](https://docs.snowflake.com/user-guide/security-access-control-overview#label-access-control-overview-role-types) when running in singledb mode. The benefit is that an administrator might be more willing to allow a team to create temporary roles that are scoped to a specific database. The detriment is that there's still the overhead of having to request an administrator to create the database role and grant it permission to create other roles.

**Additional context**
Thank you for all the time and effort you've put into making this project! My team has been trying to leverage a DCM solution within our processes and SnowDDL is perfect for our use case.

**Comment by littleK0i** [comment]
Did you try to run without `--env-prefix, but use `--target-db` with prefix baked in instead?

Example: `--source-db=PAYMENTS_DB --target-db=TED__PAYMENTS_DB`

Since there are no account-level objects to manage in SingleDB mode, `--env-prefix` is probably not needed in this mode outside of automated testing running in parallel.

**Comment by TedCha** [comment]
Thanks for the quick response! I was playing around with `--target-db` and it might be a workaround for our development flow. To give a little more context; we own two databases (ex: PROD_DB and TEST_DB). For our web application that reads data from Snowflake, the prod instance reads from PROD_DB and the test instance reads from TEST_DB. 

I'd like to use environment prefix in TEST_DB to allow for the creation of our datasets without overwriting the real datasets during unit testing and local development. Though alternatively we could just overwrite TEST_DB objects and that would be okay because there's no guarantee of stability within a test environment. We just wouldn't be able to have separated instances of objects if multiple developers are working on the same datasets or if unit tests are running (though to be honest not a big issue since we're a small team).

I'm fine with going that route if you think it's better to not introduce the complexity of `--env-prefix` within SingleDB mode.

**Comment by littleK0i** [comment]
Env prefix is applied to account-level objects only: DATABASE, USER, WAREHOUSE, etc. Env prefix is not applied to SCHEMA and schema-objects directly, but implicitly via DATABASE.

```
-- without prefix
TEST_DB.TEST_SCHEMA.TEST_TABLE

-- with prefix
PREFIX__TEST_DB.TEST_SCHEMA.TEST_TABLE
```

So using env-prefix for objects inside `TEST_DB` won't do anything.

In theory, you may create multiple test databases in advance, one per developer. But it will get messy if you have many developers working in parallel.

That's why we generally suggest to create an entire separate DEV account. You may copy some small samples of data from PROD to DEV using SHARES. It would allow you to have full flexibility in terms of testing on DEV, while keeping PROD account clean.

**Comment by TedCha** [comment]
Ahh okay. I had a fundamental misunderstanding of how environment prefixes worked. With your explanation and rereading the documentation I understand now. Thank you for the clarification and apologies for my confusion!

I agree with you completely that the ideal situation is to have a separate DEV account but the organization I work for currently doesn't have that available. Though I believe I have an idea that will allow us to keep things as clean as possible.

Feel free to close this ticket, the issue was my own misunderstanding 😅

Thanks again!
